File: app/src/main/java/com/emikhalets/datesdb/viewmodels/DateAddViewModel.java
package com.emikhalets.datesdb.viewmodels;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.work.WorkInfo;
import androidx.work.WorkManager;

import com.emikhalets.datesdb.data.AppRepository;

import java.util.UUID;

public class DateAddViewModel extends AndroidViewModel {

    private AppRepository repository;
    private MutableLiveData<String> liveDataNotice;

    public DateAddViewModel(Application application) {
        super(application);

        repository = AppRepository.getInstance(application);
        liveDataNotice = new MutableLiveData<>();
    }

    public LiveData<String> getLiveDataNotice() {
        return liveDataNotice;
    }

    public void insert(LifecycleOwner lifecycleOwner, String name, long date, String type) {
        UUID id = repository.insert(name, date, type);

        WorkManager.getInstance(getApplication())
                .getWorkInfoByIdLiveData(id)
                .observe(lifecycleOwner, info -> {
                    if (info.getState() == WorkInfo.State.SUCCEEDED) {
                        liveDataNotice.setValue("OK");
                    }
                });
    }
}

File: app/src/main/java/com/emikhalets/datesdb/viewmodels/DateAddViewModel.kt
package com.emikhalets.datesdb.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.emikhalets.datesdb.data.database.AppDatabase
import com.emikhalets.datesdb.data.database.DateItem
import com.emikhalets.datesdb.data.database.DbResult
import com.emikhalets.datesdb.data.repository.DateAddRepository
import kotlinx.coroutines.launch

class DateAddViewModel : ViewModel() {

    private val repository = DateAddRepository(AppDatabase.get().datesDao)

    private val _adding = MutableLiveData<Long>()
    val adding get(): LiveData<Long> = _adding

    private val _notice = MutableLiveData<String>()
    val notice get(): LiveData<String> = _notice

    fun insert(name: String, date: Long, type: String) {
        viewModelScope.launch {
            val dateItem = DateItem(name, date, type)
            when (val result = repository.insert(dateItem)) {
                is DbResult.Success -> _adding.postValue(result.result)
                is DbResult.Error -> _notice.postValue(result.msg)
            }
        }
    }
}